src/libYARP_os/src/yarp/os/MessageStack.cpp

```diff
@@ -124,48 +125,77 @@ class MessageStackHelper
     }
 };
 
+
 void MessageStackThread::run()
 {
     while (helper.process()) {
         // forever
     }
 }
 
-#define HELPER(x) (*((MessageStackHelper*)(x)))
 
-MessageStack::MessageStack(int max_threads)
+} // namespace
+
+
+
+class MessageStack::Private
+{
+public:
+    size_t max_threads{0};
+    MessageStackHelper* helper = nullptr;
+
+    Private(size_t max_threads) :
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'Private' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/robotology/yarp/pullRequest?prid=4590325)